server/main.py

```diff
@@ -56,115 +55,66 @@ def __init__(self) -> None:
 
         Args:
             data: The data to be sent to the clients, it should always contain a
                 "type" key to indicate the event type.
+            room: The room to which the data will be sent.
         """
-        for connection in self._active_connections:
+        for connection in self._rooms[room]:
             await connection.send(data)
 
 
 manager = ConnectionManager()
 
-html = """
-<!DOCTYPE html>
-<html>
-    <head>
-        <title>Chat</title>
-    </head>
-    <body>
-        <h1>WebSocket Chat</h1>
-        <input id="input" type="text" autocomplete="off">
-        <button onclick="sendMessage(event)">Send</button>
-        <ul id="messages"></ul>
-        <script>
-            let ws = new WebSocket("ws://localhost:8000/ws");
-
-            ws.onmessage = function(event) {
-                let data = JSON.parse(event.data);
-                switch (data.type) {
-                    case "message":
-                        let messages = document.getElementById("messages");
-                        let message = document.createElement("li");
-                        let content = document.createTextNode(data.msg);
-                        message.appendChild(content);
-                        messages.appendChild(message);
-                        break;
-                    default:
-                        console.log(`Unknown event type '${data.type}'`);
-                }
-            };
-
-            function sendMessage(event) {
-                let input = document.getElementById("input");
-                let data = {type: "message", msg: input.value};
-                ws.send(JSON.stringify(data));
-                input.value = "";
-                event.preventDefault();
-            }
-        </script>
-    </body>
-</html>
-"""
-
-
-@app.get("/")
-async def chat() -> HTMLResponse:
-    """This is the index route for the app.
-
-    Returns:
-        The HTML page in which the user can chat.
-    """
-    return HTMLResponse(html)
-
 
-@app.websocket("/ws")
-async def websocket_endpoint(websocket: WebSocket) -> None:
+@app.websocket("/ws/{room}")
+async def websocket_endpoint(websocket: WebSocket, room: str) -> None:
     """This is the endpoint for the WebSocket connection.
 
-    It creates a client and handles the connection with the ConnectionManager.
-    It continuosly receives, sends and broadcasts data to the active clients.
+    It creates a client and handles connection and disconnection with the
+    ConnectionManager. It continuously receives and broadcasts data to the
+    active clients.
     """
     client = Client(websocket)
-    await manager.connect(client)
-    await manager.broadcast({"type": "message", "msg": f"{client.id} joined the chat"})
+    await manager.connect(client, room)
 
     try:
         while True:
             data = await client.receive()
-            await manager.broadcast(data)
+            await manager.broadcast(data, room)
```

Comment:
Sending data also returns the same data to the client via the broadcast. Shouldn't it only be broadcasting to others in the room?

Reply:
Probably yes, I did this just for simplicity. Take the example of the text editor: when the client sends the event for adding a new character, I am not updating the interface straight away, but instead, I am broadcasting it to every other connected client (including the one that sent it), so when each client receives the event, it will update the interface.

Reply:
I see, and this is just for the /room/ endpoint. Regarding that, does the endpoint name need to be `/ws/{room}/`? Might also want to change the function name, too.

Reply:
Oh, good point. The route doesn't need to contain `ws`, also `websocket_endpoint` is such an awful name for the endpoint. :sweat_smile: 

Suggestions for the route and for the endpoint's name are appreciated.

Reply:
I assume we will need a few different routes, right?

1 for each of these?
![image](https://user-images.githubusercontent.com/68163698/180619188-0ce3eb30-8c14-4186-ad8a-ba01ba384252.png)

Perhaps just `connect` for actually loading the client. `connect_to_room` for starting a new room and joining an existing one since the logic is already implemented for doing that

Reply:
They won't be routes, but event types (basically a `type` key in the data that is going to be sent).

Endpoints allow the connection for a client, we shouldn't need more than one.

Reply:
Ok, that makes sense. Something like `connect_to_room` seems like a self-explanatory function then

Reply:
Names changed with https://github.com/Vthechamp22/kindly-kappa/pull/13/commits/06eec1ee0f8f7f754d141f3ca0bba796ba0c373a .